mavros_extras/src/plugins/gps_rtk.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @brief GPS RTK plugin
+ * @file gps_rtk.cpp
+ * @author Alexis Paques <alexis.paques@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2018 Alexis Paques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros_msgs/RTCM.h>
+#include <algorithm>
+
+namespace mavros {
+namespace std_plugins {
+/**
+ * @brief GPS RTK plugin
+ *
+ * Publish the RTCM messages from ROS to the Pixhawk
+ */
+class GpsRtkPlugin : public plugin::PluginBase {
+public:
+	GpsRtkPlugin() : PluginBase(),
+		gps_rtk_nh("~gps_rtk")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+		gps_rtk_sub = gps_rtk_nh.subscribe("send_rtcm", 10, &GpsRtkPlugin::rtk_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return {};
+	}
+
+private:
+	ros::NodeHandle gps_rtk_nh;
+	ros::Subscriber gps_rtk_sub;
+
+	void rtk_cb(const mavros_msgs::RTCM::ConstPtr &msg)
```

Comment:
can you please add `doxygen` compatible documentation to clarify the msg purpose and details? examples can be found in the `imu` plugin

Reply:
Also shouldn't be this called `rtcm_cb`? The reason I question this is because you actually have https://mavlink.io/en/messages/common.html#GPS_RTK, so this could actually have an handler by itself in this same plugin.

Reply:
This is right, I'll change it. 

This plugin should also be extended to subscribe to the GPS_RTK MAVLink message, maybe not in the first iteration.